dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -1646,17 +1700,21 @@ DomainParticipantImpl::enable()
   Discovery_rch disco = TheServiceParticipant->get_discovery(domain_id_);
 
   if (disco.is_nil()) {
-    ACE_ERROR((LM_ERROR,
-               ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::enable, ")
-               ACE_TEXT("no discovery found for domain id: %d.\n"), domain_id_));
+    if (DCPS_debug_level > 0) {
+      ACE_ERROR((LM_ERROR,
+                ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::enable, ")
+                ACE_TEXT("no discovery found for domain id: %d.\n"), domain_id_));
+    }
     return DDS::RETCODE_ERROR;
   }
 
 #ifdef OPENDDS_SECURITY
   if (TheServiceParticipant->get_security() && !security_config_) {
-    ACE_ERROR((LM_ERROR,
-               ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::enable, ")
-               ACE_TEXT("DCPSSecurity flag is set, but unable to load security plugin configuration.\n")));
+    if (DCPS_debug_level > 0) {
+      ACE_ERROR((LM_ERROR,
+                ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::enable, ")
+                ACE_TEXT("DCPSSecurity flag is set, but unable to load security plugin configuration.\n")));
+    }
```

Comment:
This one and a lot of the ones below it are security related errors, so they should really be controlled by an appropriate `security_debug` variable.

Reply:
Which level to be used? Maybe all security logging needs to be reviewed at some point?

Reply:
Yeah, it's a long term goal of mine to get the security logging and really all logging under tighter control. Looking at the [security logging tags](https://github.com/objectcomputing/OpenDDS/blob/5d1f84624d60f8a989e0d5d68e69c94d34b0c6e4/dds/DCPS/debug.h#L66-L84) it doesn't look like there is an appropriate one in this file, but that's okay since it's intended to be able to add new ones relatively easily. Besides here and debug.h, there are only a few places it needs to be added in debug.cpp. Maybe something like `entity_init`. It could also be two separate ones for the two functions.

Reply:
For TAOX11/AXCIOMA we have a new logger that uses ACE internally but uses log modules which can be enabled/disabled independently and uses the insertion operations with optional log formatters. It uses C++11 features so maybe something to look at for OpenDDS4?

That makes it much easier to perform logging, no %c/%d usage anymore, see for example https://github.com/RemedyIT/taox11/tree/master/tao/x11/logger and https://github.com/RemedyIT/taox11/blob/master/tests/log_test/log_test.cpp

Reply:
What is now the next best step, for our usage we would like OpenDDS to be silent, do you make a general issue for checking all security logging at a later point in time?

Reply:
> It uses C++11 features so maybe something to look at for OpenDDS4?

We were already planning on heading in that general direction.

> What is now the next best step, for our usage we would like OpenDDS to be silent, do you make a general issue for checking all security logging at a later point in time?

I think you are free to make an issue, but it's going to be a while before we can truly address this and make it possible to silence OpenDDS without silencing ACE.

Reply:
When we find more problematic areas we will make more PRs so that we can move forward with our effort

Reply:
This isn't resolved yet

Reply:
Changing them to use DCPS::security_debug.auth_warn, only in this file, will keep this in mind when I update more later on